### PR TITLE
MNT: drop 3.5 testing for 3.1 branch

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

As per dev-list email, dropping 3.5 tests from Appveyor and Travis...

Feel free to push or close this if I did it incorrectly!

Probably todo is Circle-ci.  Do we want to add 3.7 to that?  

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by tacaswell** [comment]
The 3.5 test was pinning all of the minimum deps, we need to move those over to one of the other tests.

File: .travis.yml [hunk 1]
```diff
@@ -68,11 +68,6 @@ env:
 
 matrix:
   include:
-    - python: 3.5
-      dist: trusty
-      # pytest-cov>=2.3.1 due to https://github.com/pytest-dev/pytest-cov/issues/124.
-      env:
-        - PINNEDVERS='-c requirements/testing/travis35.txt'
```
Comment (tacaswell) [inline .travis.yml]: Also remove this file (or rename it to be the 3.6 pins).

Comment (jklymak) [reply .travis.yml]: We were pinning numpy to 1.10 on the other builds?

Comment (jklymak) [reply .travis.yml]: It looks like the Travis 3.6 build was getting reqs from travis_all.txt, so I think this is OK...

Comment (jklymak) [reply .travis.yml]: OK, see new file.  I'm not sure if the other pinned reqs in travis35.txt were required to get it to run on 3.5 or if they were min deps we were wanting to test.  I suspect the former, so I removed them.

Comment (QuLogic) [reply .travis.yml]: pytest and sphinx are pinned as minimum deps also.

Comment (jklymak) [reply .travis.yml]: Why is that?  Are we testing that devs can use old Sphinx and pytest?  Or are those user facing pins?

Comment (QuLogic) [reply .travis.yml]: I suppose so; those are the versions we specify in the docs.

Comment (jklymak) [reply .travis.yml]: Fixed, - wasn't sure if w need the pin on pytest-timeout so trying w/o

**Comment by jklymak** [comment]
Looks done to me, but let me know if I missed something!

**Comment by jklymak** [comment]
Sorry @timhoffm and @tacaswell - I messed up the rebase and had to do again.  Clearing your approvals pending a new quick re-review

**Comment by jklymak** [comment]
Should there be a `ci/circleci:docs-python37`?  I'm not really clear on the point of running two doc builds.

**Comment by jklymak** [comment]
Fixed now - sorry for the confusion!

File: .circleci/config.yml [hunk 2]
```diff
@@ -121,39 +121,6 @@ jobs:
           name: "Deploy new docs"
           command: ./.circleci/deploy-docs.sh
 
-  docs-python35:
```
Comment (anntzer) [inline .circleci/config.yml]: Let's move to a 3.7 docker build, and pin numpy 1.11 on the 3.6 build?
The CircleCI build does complement the coverage of the test suite.

File: requirements/testing/travis36minver.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Extra pip requirements for the first travis python 3.6 build
```
Comment (anntzer) [inline requirements/testing/travis36minver.txt]: Needs to delete travis35.txt no?

**Comment by tacaswell** [comment]
pushed one more commit to (I think) get the circle 37 job to actually run!

**Comment by jklymak** [comment]
Thanks @tacaswell.  Not sure I understand the codecov complaint...

**Comment by jklymak** [comment]
I wonder if the azure `pip install` requirements need to be tweaked?  And should the azure 3.6 test pin numpy to 1.11?

**Comment by tacaswell** [comment]
As mentioned on the call we need to look at moving all the tests over to azure so I'm less worried about getting it's pinnings right yet.

**Comment by tacaswell** [comment]
In `test_animation` we no longer hit the 'false' path on that conditional.  Not sure on the other one yet but I suspect it is something similar.

**Comment by jklymak** [comment]
Fixed `test_animation`.  No idea how to fix `backend_ps`...

**Comment by jklymak** [comment]
Fixed itself :wink:

**Pull request merged**